src/compilation_context.ml

```diff
@@ -104,6 +106,16 @@ let create ~super_context ~scope ~expander ~obj_dir
     *)
     Sandbox_config.no_sandboxing
   in
+  let modes = lazy (
```

Comment:
It feels like this lazy is not necessary since it is systematically forced at toplevel in `Lib_rules`